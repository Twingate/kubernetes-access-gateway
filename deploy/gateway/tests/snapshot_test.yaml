suite: Snapshot
excludeTemplates:
  # TLS certificates are dynamic so these templates are tested separately.
  - NOTES.txt
  - tls-secret.yaml
tests:
  - it: should render
    set:
      metrics:
        podMonitor:
          enabled: true
    asserts:
      - matchSnapshot: {}
