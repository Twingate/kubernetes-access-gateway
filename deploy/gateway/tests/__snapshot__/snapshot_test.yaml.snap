should render:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: RELEASE-NAME-kubernetes-access-gateway-cluster-role
    rules:
      - apiGroups:
          - ""
        resources:
          - users
          - groups
          - serviceaccounts
        verbs:
          - impersonate
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: RELEASE-NAME-kubernetes-access-gateway-cluster-role-binding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: RELEASE-NAME-kubernetes-access-gateway-cluster-role
    subjects:
      - kind: ServiceAccount
        name: RELEASE-NAME-kubernetes-access-gateway
        namespace: NAMESPACE
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kubernetes-access-gateway
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: kubernetes-access-gateway-0.1.0
      name: RELEASE-NAME-kubernetes-access-gateway
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kubernetes-access-gateway
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: kubernetes-access-gateway
            app.kubernetes.io/version: 0.1.0
            helm.sh/chart: kubernetes-access-gateway-0.1.0
        spec:
          containers:
            - env:
                - name: TWINGATE_NETWORK
                  value: null
                - name: HOST
                  value: twingate.host
              image: twingate/kubernetes-gateway:latest-local
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: https
                  scheme: HTTPS
                initialDelaySeconds: 5
                periodSeconds: 5
              name: kubernetes-access-gateway
              ports:
                - containerPort: 8443
                  name: https
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /healthz
                  port: https
                  scheme: HTTPS
                initialDelaySeconds: 5
                periodSeconds: 5
              resources: {}
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1000
              volumeMounts:
                - mountPath: /etc/tls-secret-volume
                  name: tls-secret-volume
                  readOnly: true
          securityContext:
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kubernetes-access-gateway
          volumes:
            - name: tls-secret-volume
              secret:
                secretName: RELEASE-NAME-kubernetes-access-gateway-tls
  4: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations: null
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kubernetes-access-gateway
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: kubernetes-access-gateway-0.1.0
      name: RELEASE-NAME-kubernetes-access-gateway
    spec:
      ports:
        - name: https
          port: 443
          protocol: TCP
          targetPort: https
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: kubernetes-access-gateway
      type: ClusterIP
  5: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kubernetes-access-gateway
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: kubernetes-access-gateway-0.1.0
      name: RELEASE-NAME-kubernetes-access-gateway
